pulp_container/app/registry_api.py

```diff
@@ -954,23 +950,7 @@ def put(self, request, path, pk=None):
             layers = content_data.get("layers")
             blobs = set()
             for layer in layers:
-                media_type = layer.get("mediaType")
-                urls = layer.get("urls")
                 digest = layer.get("digest")
-                if (
```

Comment:
i know this is not explicit, but we need to keep this check https://github.com/quay/quay/blob/ef91c57c233e9d23372afac34c0d062a38171ffa/image/docker/schema2/manifest.py#L165
To be honest i don't understand why `urls` are not required when blob is foreign in the schema validation
maybe here you can add `urls` as additional_required if media_type == foreign_blob_types?

Reply:
Actually, this cannot be done dynamically (as you suggested: if media_type == foreign_blob_types, because we need to initialize this beforehand). I have seen conditional `required` only for keys, but not values. I am adding back the if condition.

Reply:
i see :/